rootfs/api/tests/test_config.py

```diff
@@ -849,3 +849,125 @@ def test_healthchecks_validations(self, mock_requests):
+
+        liveness_probe = {'healthcheck': {'livenessProbe': {'httpGet': {'port': 5000}}}}
+        resp = self.client.post(
+            '/v2/apps/{app_id}/config'.format(**locals()),
+            liveness_probe)
+        self.assertEqual(resp.status_code, 201, resp.data)
+        self.assertIn('livenessProbe', resp.data['healthcheck'])
+        self.assertEqual(
+            resp.data['healthcheck']['livenessProbe'],
+            liveness_probe['healthcheck']['livenessProbe'])
+        # check that the readiness probe is still there too!
+        self.assertIn('readinessProbe', resp.data['healthcheck'])
+        self.assertEqual(
+            resp.data['healthcheck']['readinessProbe'],
+            readiness_probe['healthcheck']['readinessProbe'])
+
+        # post a new build
+        response = self.client.post(
+            "/v2/apps/{app_id}/builds".format(**locals()),
+            {'image': 'quay.io/autotest/example'}
+        )
+        self.assertEqual(response.status_code, 201, response.data)
+
+    def test_config_healthchecks_validations(self, mock_requests):
+        """
+        Test that healthchecks validations work
+        """
+        response = self.client.post('/v2/apps')
+        self.assertEqual(response.status_code, 201, response.data)
+        app_id = response.data['id']
+
+        # Set one of the values that require a numeric value to a string
+        resp = self.client.post(
+            '/v2/apps/{app_id}/config'.format(**locals()),
+            {'healthcheck': json.dumps({'livenessProbe':
+                                        {'httpGet': {'port': '50'}, 'initialDelaySeconds': "t"}})}
+        )
+        self.assertEqual(resp.status_code, 400, response.data)
+
+        # Don't set one of the mandatory value
+        resp = self.client.post(
+            '/v2/apps/{app_id}/config'.format(**locals()),
+            {'healthcheck': json.dumps({'livenessProbe':
+                                        {'httpGet': {'path': '/'}, 'initialDelaySeconds': 1}})}
+        )
+        self.assertEqual(resp.status_code, 400, response.data)
+
+    def test_config_healthchecks_legacy(self, mock_requests):
+        """
+        Test that when a user uses `deis config:set HEALTHCHECK_URL=/`, the config
+        object is rolled over to the `healthcheck` field.
+        """
+        response = self.client.post('/v2/apps')
+        self.assertEqual(response.status_code, 201, response.data)
+        app = App.objects.get(id=response.data['id'])
+
+        # Set healthcheck URL to get defaults set
+        response = self.client.post(
+            '/v2/apps/{app.id}/config'.format(**locals()),
+            {'values': json.dumps({'HEALTHCHECK_URL': '/health'})}
+        )
+        self.assertEqual(response.status_code, 201, response.data)
+        self.assertIn('HEALTHCHECK_URL', response.data['values'])
+        self.assertEqual(response.data['values']['HEALTHCHECK_URL'], '/health')
+        # legacy defaults
```

Comment:
A note I should bring up... Are these really the defaults we should be using? I referenced them based off of https://github.com/deis/controller/blob/master/rootfs/api/models/config.py#L40-L45 but the scheduler has conflicting options.